main.go

```diff
@@ -441,18 +464,48 @@ func (svr *bridge) handleCall(w http.ResponseWriter, r *http.Request) {
 }
 
 func renderTemplate(templateString string, data interface{}) (string, error) {
-	titleTemplate, err := template.New("tmp").Parse(templateString)
-	if err != nil {
-		return "", fmt.Errorf("error in Template: %s", err)
+	var result string
+	var err error
+	var unsupportedFunc string
+
+	//Excludes unsupported template function calls.
+	switch {
+	case (strings.Contains(templateString, "{{ query") || strings.Contains(templateString, "{{query")):
+		unsupportedFunc = "query"
+	case (strings.Contains(templateString, "{{ first") || strings.Contains(templateString, "{{first")):
+		unsupportedFunc = "first"
+	case (strings.Contains(templateString, "{{ label") || strings.Contains(templateString, "{{label")):
+		unsupportedFunc = "label"
+	case (strings.Contains(templateString, "{{ value") || strings.Contains(templateString, "{{value")):
+		unsupportedFunc = "value"
+	case (strings.Contains(templateString, "{{ strvalue") || strings.Contains(templateString, "{{strvalue")):
+		unsupportedFunc = "strvalue"
+	case (strings.Contains(templateString, "{{ safeHtml") || strings.Contains(templateString, "{{safeHtml")):
+		unsupportedFunc = "safeHtml"
+	case (strings.Contains(templateString, "{{ graphLink") || strings.Contains(templateString, "{{graphLink")):
+		unsupportedFunc = "graphLink"
+	case (strings.Contains(templateString, "{{ sortByLabel") || strings.Contains(templateString, "{{sortByLabel")):
+		unsupportedFunc = "sortByLabel"
+	case (strings.Contains(templateString, "{{ pathPrefix") || strings.Contains(templateString, "{{pathPrefix")):
+		unsupportedFunc = "pathPrefix"
+	case (strings.Contains(templateString, "{{ externalURL") || strings.Contains(templateString, "{{externalURL")):
+		unsupportedFunc = "externalURL"
+	default:
+		unsupportedFunc = ""
 	}
 
-	var templatedTitle bytes.Buffer
-	err = titleTemplate.ExecuteTemplate(&templatedTitle, "tmp", data)
-	if err != nil {
-		return "", fmt.Errorf("error in Template: %s", err)
-	}
+	if unsupportedFunc == "" {
+		titleTemplate := template.NewTemplateExpander(context.Background(), templateString, "tmp", data, 0, nil, nil, nil)
+		result, err = titleTemplate.Expand()
+		if err != nil {
+			return "", fmt.Errorf("error in Template: %s", err)
+		}
 
-	return templatedTitle.String(), nil
+		result = strings.ReplaceAll(result, "<no value>", "[no value]")
```

Comment:
Looking through the source code for [template](https://github.com/prometheus/prometheus/blob/main/template/template.go#L143-L145), it looks like this shouldn't be needed since the default is to set `missingkey=zero` - which should avoid the injection of `<no value>`. Have you observed `<no value>` showing up in output?

Reply:
Thanks for catching this. This line is not required. I forgot to remove it when I was reverse-engineering some code.